frontend/src/hooks/useUserSession.js

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+import { useExceptionHandler } from "../hooks/useExceptionHandler.jsx";
+import { useAlertStore } from "../store/alert-store";
+const useUserSession = () => {
+  const handleException = useExceptionHandler();
+  const { setAlertDetails } = useAlertStore();
+  return async () => {
+    try {
+      const requestOptions = {
+        method: "GET",
+        url: "/api/v1/session",
+      };
+      const res = await axios(requestOptions);
+      return res.data;
+    } catch (error) {
+      if (error?.response?.statusText === "Payment Required") {
```

Comment:
@athul-rs Suggestion for cleaner code. Please feel free to ignore if you think otherwise.

```
  const fallbackErrMsg = "Error while getting session";
  if (error?.response?.statusText === "Payment Required") {
    setAlertDetails(handleException(error, fallbackErrMsg));
    return;
  }
  
  if (error?.response?.statusText === "Unauthorized") {
    setAlertDetails(handleException(error, fallbackErrMsg));
    return;
  }

  setAlertDetails(handleException(error, fallbackErrMsg));
```

Reply:
resolved